modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/CreateTableCommand.java

```diff
@@ -102,7 +110,14 @@ private CreateTableCommand(
     public List<UpdateEntry> get(Catalog catalog) {
         CatalogSchemaDescriptor schema = schemaOrThrow(catalog, schemaName);
 
-        ensureNoTableIndexOrSysViewExistsWithGivenName(schema, tableName);
+        int checkType = ignoreErrorIfExist ? SYS_VIEW | INDEX : ALL;
+
+        ensureNoTableIndexOrSysViewExistsWithGivenName(schema, tableName, checkType);
+
+        // table already created, but probably current schema still not active
+        if (ignoreErrorIfExist && schema.table(tableName) != null) {
+            return List.of();
```

Comment:
I'm still not sure it's best to always equalize 'no entries' with 'do not create because it already exists AND we were asked to be quiet because of IF NOT EXISTS'. Is an explicit flag for this ('did not create anything as it's already created') in the return value an option?